nimbus-core/src/main/java/com/antheminc/oss/nimbus/integration/mq/ActiveMqConsumer.java

```diff
@@ -0,0 +1,81 @@
+/**
+ *  Copyright 2016-2019 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.antheminc.oss.nimbus.integration.mq;
+
+import java.io.IOException;
+
+import org.springframework.jms.annotation.JmsListener;
+
+import com.antheminc.oss.nimbus.FrameworkRuntimeException;
+import com.antheminc.oss.nimbus.domain.cmd.CommandBuilder;
+import com.antheminc.oss.nimbus.domain.cmd.CommandMessage;
+import com.antheminc.oss.nimbus.domain.cmd.exec.CommandExecutorGateway;
+import com.antheminc.oss.nimbus.support.JustLogit;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+/**
+ * @author Sandeep Mantha
+ * @author Tony Lopez
+ */
+@Getter
+@Setter
+@RequiredArgsConstructor
+public class ActiveMqConsumer {
+
+	public static final JustLogit LOG = new JustLogit(ActiveMqConsumer.class);
+
+	private final CommandExecutorGateway executorGateway;
```

Comment:
Please refer to WebCommandDispatcher and see if a similar strategy can be used, so that in future other Queue/Topic based listeners can use the same Dispatcher to delegate to CmdGateway.

Reply:
Fixed.